sale_commission_advance/models/account_move.py

```diff
@@ -0,0 +1,129 @@
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def create(self, vals):
+        rec = super(AccountMove, self).create(vals)
+        if rec.has_related_sale_with_down_payment():
+            rec.recompute_lines_agents()
+        return rec
+
+    def has_related_sale_with_down_payment(self):
+        if any(
+            self.invoice_line_ids.mapped(
+                lambda x: x.sale_line_ids and x.sale_line_ids.is_downpayment
+            )
+        ):
+            return True
+        return False
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends("move_id.partner_id")
+    def _compute_agent_ids(self):
+        self.agent_ids = False  # for resetting previous agents
+        for record in self.filtered(
+            lambda x: x.move_id.partner_id and x.move_id.move_type[:3] == "out"
+        ):
+            if record.sale_line_ids and record.sale_line_ids.is_downpayment:
```

Comment:
No need of the double if, as if no sale line, it will be False as well.

```suggestion
            if record.sale_line_ids.is_downpayment:
```